File: PracticasProgramacion/Practica_10/python/estadisticas_arreglo.py
"""
EstadisticasArreglo: datos en lista fija de 5 elementos.
Métodos get_* realizan el cálculo sobre self._numeros.
"""


class EstadisticasArreglo:
    N = 5

    def __init__(self) -> None:
        self._numeros = [0.0] * self.N

    def ingresar_numeros(self) -> None:
        print("Ingrese 5 numeros en el arreglo:")
        for i in range(self.N):
            self._numeros[i] = float(input(f"numeros[{i}]: "))

    def get_suma(self) -> float:
        return sum(self._numeros)

    def get_promedio(self) -> float:
        return self.get_suma() / self.N

    def get_mediana(self) -> float:
        c = sorted(self._numeros)
        return c[self.N // 2]

    def get_maximo(self) -> float:
        return max(self._numeros)

    def get_minimo(self) -> float:
        return min(self._numeros)

    def mostrar_resultados(self) -> None:
        print(f"\nSuma: {self.get_suma()}")
        print(f"Promedio: {self.get_promedio()}")
        print(f"Mediana: {self.get_mediana()}")
        print(f"Maximo: {self.get_maximo()}")
        print(f"Minimo: {self.get_minimo()}")

File: PracticasProgramacion/Practica_10/cpp/EstadisticasArreglo.cpp
#include "EstadisticasArreglo.h"
#include <iostream>
#include <algorithm>
#include <numeric>

void EstadisticasArreglo::ingresarNumeros() {
    std::cout << "Ingrese 5 numeros en el arreglo:\n";
    for (int i = 0; i < N; ++i) {
        std::cout << "numeros[" << i << "]: ";
        std::cin >> numeros_[i];
    }
}

void EstadisticasArreglo::calcular() const {
    // Metodos get hacen los calculos
}

void EstadisticasArreglo::mostrarResultados() const {
    std::cout << "\nSuma: " << getSuma() << "\n";
    std::cout << "Promedio: " << getPromedio() << "\n";
    std::cout << "Mediana: " << getMediana() << "\n";
    std::cout << "Maximo: " << getMaximo() << "\n";
    std::cout << "Minimo: " << getMinimo() << "\n";
}

double EstadisticasArreglo::getSuma() const {
    return std::accumulate(numeros_, numeros_ + N, 0.0);
}
double EstadisticasArreglo::getPromedio() const {
    return getSuma() / N;
}
double EstadisticasArreglo::getMediana() const {
    double copia[N];
    for (int i = 0; i < N; ++i) copia[i] = numeros_[i];
    std::sort(copia, copia + N);
    return copia[N/2];
}
double EstadisticasArreglo::getMaximo() const {
    return *std::max_element(numeros_, numeros_ + N);
}
double EstadisticasArreglo::getMinimo() const {
    return *std::min_element(numeros_, numeros_ + N);
}
